src/main/java/io/nats/service/InfoResponse.java

```diff
@@ -33,13 +36,21 @@ public class InfoResponse extends ServiceResponse {
     private final String description;
     private final List<Endpoint> endpoints;
 
-    InfoResponse(String id, String name, String version, Map<String, String> metadata, String description, Collection<ServiceEndpoint> serviceEndpoints) {
+    InfoResponse(String id, String name, String version, Map<String, String> metadata, String description) {
         super(TYPE, id, name, version, metadata);
         this.description = description;
         this.endpoints = new ArrayList<>();
-        for (ServiceEndpoint se : serviceEndpoints) {
-            addServiceEndpoint(se);
-        }
+    }
+
+    InfoResponse addServiceEndpoint(ServiceEndpoint se) {
+        endpoints.add(new Endpoint(
+            se.getName(),
+            se.getSubject(),
+            se.getQueueGroup(),
+            se.getMetadata()
+        ));
+        serialized.set(null);
+        return this;
```

Comment:
1. this method should not be public. Only the service code is supposed to build this.
2. Clear the lazy serialization. when an endpoint is added